pkg/controller/apigateway/restapi/setup.go

```diff
@@ -0,0 +1,273 @@
+	}
+
+	if cr.Spec.ForProvider.Policy != nil {
+		pol, err := normalizePolicy(cr.Spec.ForProvider.Policy)
+		if err != nil {
+			return errors.Wrap(err, "cant normalize managed resource policy")
+		}
+
+		cr.Spec.ForProvider.Policy = pol
+	}
+
+	pOps, err := apigwclient.GetPatchOperations(&cur, &cr.Spec.ForProvider)
+	if err != nil {
+		return errors.Wrap(err, "cant compute patch preUpdate")
+	}
+
+	obj.PatchOperations = pOps
+	obj.RestApiId = aws.String(meta.GetExternalName(cr))
+
+	return nil
+}
+
+func policiesAreKindOfTheSame(a *string, b *string) (bool, error) {
+	if a != nil && b != nil {
+		aPol, bPol, err := parsePolicies(a, b)
+		if err != nil {
+			return false, errors.Wrap(err, "cant parse policies")
+		}
+		polPatch, err := apigwclient.GetJSONPatch(aPol, bPol)
+		if err != nil {
+			return false, errors.Wrap(err, "cant compute jsonpatch")
+		}
+
+		for _, p := range polPatch {
+			re := regexp.MustCompile(`/Statement/(\d+)/Resource`)
+			if p.Operation != "replace" || !re.MatchString(p.Path) {
+				return false, nil
+			}
+
+			index, _ := strconv.Atoi(re.FindString(p.Path))
+			fmt.Println(aPol["Statement"].([]interface{})[index].(map[string]interface{})["Resource"])
+			if aPol["Statement"].([]interface{})[index].(map[string]interface{})["Resource"] != "execute-api:/*/*/*" {
+				return false, nil
+			}
+		}
+
+		return true, nil
+	}
+
+	return cmp.Equal(a, b), nil
+}
+
+func preDelete(_ context.Context, cr *svcapitypes.RestAPI, obj *svcsdk.DeleteRestApiInput) (bool, error) {
+	obj.RestApiId = aws.String(meta.GetExternalName(cr))
+	return false, nil
+}
+
+func isUpToDate(cr *svcapitypes.RestAPI, cur *svcsdk.RestApi) (bool, error) {
+	s := &svcapitypes.RestAPIParameters{
+		Name:   cur.Name,
+		Region: cr.Spec.ForProvider.Region,
+	}
+
+	if cur.Policy != nil {
+		cur.Policy = unescapePolicy(cur.Policy)
+	}
+
+	if err := lateInitialize(s, cur); err != nil {
+		return false, errors.Wrap(err, "cant lateinit")
+	}
+
+	patchJSON, err := aws.CreateJSONPatch(cr.Spec.ForProvider, &s)
+	if err != nil {
+		return false, errors.Wrap(err, "error checking up to date")
+	}
+
+	patch := &svcapitypes.RestAPIParameters{}
+	if err := json.Unmarshal(patchJSON, patch); err != nil {
+		return false, errors.Wrap(err, "error checking up to date")
+	}
+
+	return cmp.Equal(&svcapitypes.RestAPIParameters{}, patch,
+		cmpopts.IgnoreTypes([]xpv1.Reference{}, []xpv1.Selector{}),
+		cmpopts.IgnoreFields(svcapitypes.RestAPIParameters{}, "Region"),
+	), nil
+}
+
+func normalizePolicy(p *string) (*string, error) {
+	if p == nil {
+		return p, nil
+	}
+	var mappedPol map[string]interface{}
+	if err := json.Unmarshal([]byte(*p), &mappedPol); err != nil {
+		return nil, errors.Wrap(err, "cant unmarshal policy")
+	}
+
+	parsed, err := json.Marshal(mappedPol)
+	if err != nil {
+		return nil, err
+	}
+
+	return aws.String(string(parsed)), nil
+}
+
+func unescapePolicy(p *string) *string {
+	if p == nil {
+		return p
+	}
+
+	s := strings.ReplaceAll(*p, "\\\"", "\"")
```

Comment:
I'm not sure why we are doing a string replace here to unescape the data coming in. According to the AWS API it should be stringified data.

Reply:
This was added because policies come from AWS in the format {\"Version\","\Statement\"}, which did not play well with json.Unmarshal (which I needed for upToDate deep comparison). I'd love to take this out but didn't know a better way to do this.